pkg/policy/l4.go

```diff
@@ -786,7 +786,16 @@ func addL4Filter(policyCtx PolicyContext,
 		filterToMerge.detach(selectorCache)
 		return err
 	}
-	existingFilter.DerivedFromRules = append(existingFilter.DerivedFromRules, ruleLabels)
+	var exists bool
+	for _, existingRules := range existingFilter.DerivedFromRules {
```

Comment:
I'd have "labels" somewhere in the variable name, like `existingRuleLabels` to make it clearer we not somehow comparing a "rule" to "labels" below.